### PR TITLE
Add async approach to import course from scorm cloud 

### PR DESCRIPTION
**Pull request by victoriawugrovo**
Add async approach to import course from scorm cloud using importCourseAsync and getAsyncImportResult.
Doc here : https://cloud.scorm.com/docs/api_reference/course.html

### CODE REVIEW

File: src/CourseService.php [hunk 1]
```diff
@@ -386,6 +386,89 @@ public function UpdateAssetsFromUploadedFile($courseId, $uploadLocation)
...
     	return $response;
     }
+
+    /// <summary>
+    /// Import a SCORM .pif (zip file) from the local filesystem using async client.
+    /// </summary>
+    /// <param name="courseId">Unique Identifier for this course.</param>
+    /// <param name="absoluteFilePathToZip">Full path to the .zip file</param>
+    /// <returns>List of Import Results</returns>
+    public function ImportCourseAsync($courseId, $absoluteFilePathToZip)
```
Comment (pranavgarg) [inline src/CourseService.php]: typehint the response

File: src/CourseService.php [hunk 2]
```diff
@@ -386,6 +386,89 @@ public function UpdateAssetsFromUploadedFile($courseId, $uploadLocation)
...
+        $importException = null;
+        $response = null;
+        try {
+            $token = $this->ImportCourseAsyncToken($courseId, $location);
+            $done = false;
+            while (!$done){
+                $xmlStatus = $this->getAsyncImportStatus($token);
+                $statusResult = (string) $xmlStatus->status;
+                switch ($statusResult) {
+                    case 'running':
```
Comment (pranavgarg) [inline src/CourseService.php]: make these as const.

Comment (pranavgarg) [reply src/CourseService.php]: so is response an array or an object?

File: src/CourseService.php [hunk 3]
```diff
@@ -386,6 +386,89 @@ public function UpdateAssetsFromUploadedFile($courseId, $uploadLocation)
...
+        return $response;
+    }
+
+    /// <summary>
+    /// Get import course token
+    /// </summary>
+    /// <param name="courseId">Unique Identifier for the course</param>
+    /// <param name="path">The relative path to the uploaded zip file</param>
+    /// <returns>Token Id</returns>
+    public function ImportCourseAsyncToken($courseId, $path)
```
Comment (pranavgarg) [inline src/CourseService.php]: typehint the return value of the function.

File: src/CourseService.php [hunk 4]
```diff
@@ -386,6 +386,89 @@ public function UpdateAssetsFromUploadedFile($courseId, $uploadLocation)
...
+    /// </summary>
+    /// <param name="token">Unique token for the course upload</param>
+    /// <returns>Import Status and Result as SimpleXMLElement</returns>
+    private function getAsyncImportStatus($token)
+    {
+        $request = new ServiceRequest($this->_configuration);
+        $params = array('token'=>$token);
+        $request->setMethodParams($params);
+        $response = $request->CallService("rustici.course.getAsyncImportResult");
+        write_log('rustici.course.getAsyncImportResult : '.$response);
```
Comment (pranavgarg) [inline src/CourseService.php:472]: where is write_log writing too? Also shouldn't we use logger?

File: src/CourseService.php [hunk 5]
```diff
@@ -386,6 +386,89 @@ public function UpdateAssetsFromUploadedFile($courseId, $uploadLocation)
...
+    /// Get import course status and result using async client
+    /// </summary>
+    /// <param name="token">Unique token for the course upload</param>
+    /// <returns>Import Status and Result as SimpleXMLElement</returns>
+    private function getAsyncImportStatus($token)
+    {
+        $request = new ServiceRequest($this->_configuration);
+        $params = array('token'=>$token);
+        $request->setMethodParams($params);
+        $response = $request->CallService("rustici.course.getAsyncImportResult");
```
Comment (pranavgarg) [inline src/CourseService.php:471]: is this a method we are calling from the library or somehow attaching to the library?

**Review by pranavgarg — CHANGES_REQUESTED** [review]
Do we have test cases as to test the behavior of the functions?

File: src/CourseService.php (hunk 4 above)
Comment (victoriawugrovo) [reply src/CourseService.php:472]: The whole library always uses write_log, you want me to change the logger for the whole library?

File: src/CourseService.php (hunk 2 above)
Comment (victoriawugrovo) [reply src/CourseService.php]: You mean the response from the web service? Then it is an xml, not an object. I have to convert it to xml element to fetch the status

File: src/CourseService.php (hunk 1 above)
Comment (victoriawugrovo) [reply src/CourseService.php]: typehint is disallowed due the version of php. Would you like me to upgrade?

File: src/CourseService.php (hunk 5 above)
Comment (victoriawugrovo) [reply src/CourseService.php:471]: The method is attached in the library. Here: https://github.com/GrovoLearning/SCORMCloud_PHPLibrary/blob/master/src/ServiceRequest.php. But the whole library we forked from someone else'

Comment (pranavgarg) [reply src/CourseService.php:471]: can you check if the existing deployed code is writing these messages

File: src/CourseService.php (hunk 4 above)
Comment (pranavgarg) [reply src/CourseService.php:472]: if its already working then no need to change, just ensure where these logs are getting printed is it stdout or to some file.

File: composer.json [hunk 6]
```diff
@@ -8,5 +8,8 @@
     ],
     "autoload": {
       "classmap": ["src/"]
+    },
```
Comment (pranavgarg) [inline composer.json:11]: just curious why do you have to specify the php version in the library? shouldn't it be in the parent folder in which its being used?

File: src/CourseService.php (hunk 4 above)
Comment (victoriawugrovo) [reply src/CourseService.php:472]: Logs are not printed. They should be writing to 'SCORMCloud_debug.log', https://github.com/GrovoLearning/SCORMCloud_PHPLibrary/blob/master/src/DebugLogger.php, but we disabled the debug. so it is not writing to the log file.

File: src/CourseService.php (hunk 5 above)
Comment (victoriawugrovo) [reply src/CourseService.php:471]: Existing deployed code isn't using async approach, it is using sync approach, from line 61-80 and line 98-118 here https://github.com/GrovoLearning/SCORMCloud_PHPLibrary/blob/master/src/CourseService.php

**Review by pranavgarg — APPROVED** [review]

**Pull request merged**